src/main/java/com/lothrazar/cyclic/event/ItemEvents.java

```diff
@@ -199,4 +199,31 @@ public void onHit(PlayerInteractEvent.LeftClickBlock event) {
       UtilChat.sendStatusMessage(player, UtilChat.lang(WrenchActionType.getName(held)));
     }
   }
+
+  @SubscribeEvent
+  public void onPlayerPickup(EntityItemPickupEvent event) {
```

Comment:
You mention a bug on pickup, where after the try insert into the bag, the player also gets a copy of the stack.  Would it be solved by cancelling the event when the insert is a success? either with .setCanceled(true); or .setResult(Result.DENY);

Reply:
no I got it. I was just checking and comparing the wrong stacks after insertion. You don't want to cancel the event anyways, because you want the ItemEntity to go away if you took it all and you need to `ALLOW` the event for that.